CanvasColorSpaceProposal.md

```diff
@@ -45,178 +45,172 @@ You can see this by making stripes of 1.0 and 0.0, and matching that pattern's b
-* createImageData() and getImageData() produce an ImageData object with the same color space as the source canvas, using an ImageDataArray of a type that is appropriate for the pixelFormat of the source canvas (smallest possible numeric size that guarantees no loss of precision).
-* putImageData() performs a color space conversion to the color space of the destination canvas.
-* Data returned by getImageData() or passed to putImageData() are assumed to be in "srgb" color space, with "unorm8" encoding.
+An ImageData's data attribute will be of the type indicated by the its storageFormat.
 
-### Limitations
-* toDataURL and toBlob may be lossy, depending on the file format, when used on a canvas that has an encoding other than "unorm8". Possible future improvements could solve or mitigate this issue by adding more file formats or adding options to specify the resource color space.
+Note that the storage formats are different from the pixel formats for 2D Canvas and WebGL pixel formats. This reflects the fact that there does not exist a Float16Array, although Float16 is a useful format for graphics buffers.
+
+### ImageBitmap
+
+ImageBitmaps are opaque bitmap objects that can be used by several consumers. ImageBitmaps should internally store their contents in the highest fidelity format possible.
+
+For example, if an ImageBitmap is created from a wide-color-gamut high-bit-depth image and is then drawn to a wide-color-gamut 2D canvas, the result should reflect the original image. In particular, the image should not have been clamped to sRGB or quantized to 8 bits anywhere along the way.
+
+TODO: createImageBitmap has a colorSpaceConversion flag that can be "none" or "default". The most useful behavior is "respect the color space and convert appropriately when drawing to a canvas", which is the behavior of "default" on some browsers. A less useful but sometimes desirable behavior is "ignore the color space, and reinterpret the pixels as though they were in sRGB", which is the behavior of "none". This behavior should be more clearly specified.
 
 ### Adoption
+
 Lack of color management and color interoperability is a longstanding complaint about the canvas API.
 Authors of games and imaging apps are expected to be enthusiastic adopters.
 
-## Unresolved Issues
+## Issues
 
-* Should we support custom color spaces based on ICC profiles? Would offer ultimate flexibility. Would be hard to make implementations as efficient as built-in color spaces, in particular for implement linearPixelMath for profiles that have arbitrary transfer curves.
+Issues:
 
-* Should float16 allow alpha values outside [0,1] range at any stage of the pipeline? What would they mean?
+* How should feature detection work? The combinations of CanvasColorSpace and CanvasColorType supported by 2D Canvas and WebGL are, for now, obviously "all of them", but this is not likely to remain the case.
```

Comment:
The canvas context attributes can already be queried by the user after the canvas context has been created.  If the query returns a dictionary with no `colorSpace` or `storageFormat` members, the developer can conclude they were ignored.

Reply:
If a browser were to add, say, "adobe-rgb" as a color space, how do does one detect ahead of time that that is supported? It would be nice to be able to check before creating the context.

(But I'm new in this area, so maybe that's the convention)

Reply:
@ccameron-chromium , you're right that there's no way to tell ahead of time whether there is support. 
 You have to try things and see whether the request was honored.  

ImageBitmap has [a similar problem with `ImageBitmapOptions`](https://github.com/whatwg/html/pull/4248) which hasn't been resolved.

Reply:
Thanks for the context!